tests/baryotests/Test-baryo-c2hdm.cpp

```diff
@@ -0,0 +1,104 @@
+// SPDX-FileCopyrightText: 2021 Philipp Basler, Margarete Mühlleitner and Jonas
+// Müller
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "catch.hpp"
+
+#include <BSMPT/baryo_calculation/CalculateEtaInterface.h>
+#include <BSMPT/minimizer/Minimizer.h>
+#include <BSMPT/models/ClassPotentialOrigin.h> // for Class_Potential_Origin
+#include <BSMPT/models/IncludeAllModels.h>
+#include <BSMPT/models/ModelTestfunctions.h>
+
+#include "GenerateTestCompares/C2HDM.h"
+
+const std::vector<double> example_point_C2HDM{/* lambda_1 = */ 3.29771,
+                                              /* lambda_2 = */ 0.274365,
+                                              /* lambda_3 = */ 4.71019,
+                                              /* lambda_4 = */ -2.23056,
+                                              /* Re(lambda_5) = */ -2.43487,
+                                              /* Im(lambda_5) = */ 0.124948,
+                                              /* Re(m_{12}^2) = */ 2706.86,
+                                              /* tan(beta) = */ 4.64487,
+                                              /* Yukawa Type = */ 1};
+
+const Compare_C2HDM Expected;
+
+TEST_CASE("Checking EWBG for C2HDM", "[c2hdm]")
+{
+  using namespace BSMPT;
+  std::shared_ptr<BSMPT::Class_Potential_Origin> modelPointer =
+      ModelID::FChoose(ModelID::ModelIDs::C2HDM);
+  modelPointer->initModel(example_point_C2HDM);
+
+  const auto WhichMin = Minimizer::WhichMinimizerDefault;
+
+  const auto EWPT = Expected.EWPTPerSetting.at(WhichMin);
+
+  std::vector<double> vevsymmetricSolution, checksym, startpoint;
+  for (const auto &el : EWPT.EWMinimum)
+    startpoint.push_back(0.5 * el);
+  vevsymmetricSolution = Minimizer::Minimize_gen_all(
+      modelPointer, EWPT.Tc + 1, checksym, startpoint, WhichMin, true);
+  auto min_expected      = Expected.vevSymmetricPerSetting.at(WhichMin);
+  const double threshold = 1e-4;
+  REQUIRE(vevsymmetricSolution.size() == min_expected.size());
+  for (std::size_t i{0}; i < vevsymmetricSolution.size(); ++i)
+  {
+    auto res      = std::abs(vevsymmetricSolution.at(i));
+    auto expected = std::abs(min_expected.at(i));
+    if (expected > threshold)
+    {
+      UNSCOPED_INFO("Current Option for Minimizer:\t" << WhichMin);
+      UNSCOPED_INFO("This ist the position:"
+                    << i << "\tFound solution =" << vevsymmetricSolution.at(i)
+                    << "\tExpected solution = " << min_expected.at(i));
+      REQUIRE(res == Approx(expected).epsilon(1e-4));
+    }
+    else
+    {
+      REQUIRE(res <= threshold);
```

Comment:
is this test independent on the chosen minimizer?

Reply:
No, but WhichMin is used both in getting the expected result and calculating the minimum